docker/Dockerfile

```diff
@@ -0,0 +1,69 @@
+FROM debian:11-slim as build
+
+WORKDIR /build
+
+RUN --mount=type=cache,target=/var/cache/apt \
+    echo "Install build dependencies" && \
+    apt update && \
+    apt install gcc cmake zlib1g-dev make build-essential g++ unzip ca-certificates wget -yqq --no-install-recommends && \
+    rm -rf /var/lib/apt/lists/* && \
+    update-ca-certificates && \
+    wget https://dev.mysql.com/get/Downloads/Connector-C++/libmysqlcppconn9_8.0.27-1debian11_amd64.deb -O /tmp/libmysqlcppconn.deb && \
+    dpkg -i /tmp/libmysqlcppconn.deb && \
+    rm /tmp/libmysqlcppconn.deb
+
+COPY dAuthServer/ /build/dAuthServer
+COPY dChatServer/ /build/dChatServer
+COPY dCommon/ /build/dCommon
+COPY dChatFilter/ /build/dChatFilter
+COPY dDatabase/ /build/dDatabase
+COPY dGame/ /build/dGame
+COPY dMasterServer/ /build/dMasterServer
+COPY dNet/ /build/dNet
+COPY dPhysics/ /build/dPhysics
+COPY dScripts/ /build/dScripts
+COPY dWorldServer/ /build/dWorldServer
+COPY dZoneManager/ /build/dZoneManager
+COPY migrations/ /build/migrations
+COPY resources/ /build/resources
+COPY thirdparty/ /build/thirdparty
+COPY vanity /build/vanity
+COPY .clang-* CMake* LICENSE /build/
+
+ARG BUILD_THREADS=1
+ARG BUILD_VERSION=171022
```

Comment:
This should probably be a value read from `.env` so that users can choose client version

Reply:
Yes, true. The issue is, BUILD_VERSION gets referenced in cmake in the build phase. This requires a change in the actual source code.

Reply:
Yeah I like the compose route. Have an args list and everything in the .env is auto read by compose